src/Enzo/enzo_EnzoBfieldMethod.hpp

```diff
@@ -0,0 +1,218 @@
+  ///
+  /// For constrained transport, this includes both the face-centered and the
+  /// cell-centered bfields (the face-centered bfield updates are handled
+  /// completely internally). The velocity and bfield components stored in this
+  /// `cur_prim_map` map are used to compute the cell-centered E-field.
+  ///
+  /// @param[in]  cur_prim_map Map containing the current values of the
+  ///     cell-centered integrable quantities (before they have been updated
+  ///     over the current timestep). 
+  /// @param[in]  xflux_map,yflux_map,zflux_map Maps containing the values of
+  ///     the fluxes computed along the x, y, and z directions. The function
+  ///     namely makes use of the various magnetic field fluxes
+  /// @param[out] out_centered_bfield_map Map holding the arrays where the
+  ///     updated values for each cell-centered magnetic field component should
+  ///     be stored. These arrays can be aliases of `cur_prim_map` (The
+  ///     arguments can even reference the same object).
+  /// @param[in]  dt The (partial) time-step over which to update the magnetic
+  ///     fields.
+  /// @param[in]  stale_depth indicates the current stale_depth for the
+  ///     supplied quantities. This should nominally be the same as the stale
+  ///     depth used to compute the fluxes and that is passed to
+  ///     EnzoIntegrableUpdate::update_quantities.
+  virtual void update_all_bfield_components
+  (EnzoEFltArrayMap &cur_prim_map, EnzoEFltArrayMap &xflux_map,
+   EnzoEFltArrayMap &yflux_map, EnzoEFltArrayMap &zflux_map,
+   EnzoEFltArrayMap &out_centered_bfield_map, enzo_float dt, int stale_depth)
+    noexcept=0;
+
+protected:
+
+  /// Virtual method that subclasses overide to preload any method specific
+  /// data from the new target block and optionally initialize scratch space
+  ///
+  /// @param[in] target_block holds data to be processed
+  /// @param[in] first_initialization Indicates if this is the first call since
+  ///    construction (including deserialization). When true, scratch space
+  ///    data should be allocated.
+  virtual void register_target_block_(Block *target_block,
+                                      bool first_initialization) noexcept = 0;
+
+  /// Helper method that raises an error if there isn't a registered block.
+  void require_registered_block_() const noexcept
+  {
+    if (target_block_ == nullptr){
+      ERROR("EnzoConstrainedTransport::require_registered_block_",
+            "An invalid method has been executed that requires that a target "
+            "block is registered.");
+    }
+  }
+
+  int num_partial_timesteps() const noexcept {return num_partial_timesteps_;}
+
+private: // keep subclasses from touching these attributes
+
+  /// Number of partial timesteps in a cycle. A value of 1 would means that
+  /// there is only a single update over the full timestep. A value of 2 means
+  /// that there is a half timestep and a full timestep.
+  int num_partial_timesteps_;
```

Comment:
Currently the integration scheme is just 2nd order accurate in time (Van Leer).

To answer your question: Yes, it was my intention to make the implementation flexible enough, that it could be easily extended to other integration orders. I think it should be trivial to modify `EnzoMethodMHDVlct` so that you could switch between the Van Leer scheme and a 3rd order integration in time scheme (we would need to change the name of the class in that case). But in the worst case scenario, you could definitely reuse all of the MHD helper classes.

As far as I can tell, we would need to make 3 main tweaks:
1. We need to be able the number of times we enter the for-loop in VL+CT and tweak our handling of the temporary arrays used for holding values at partial timesteps
2. `EnzoBfieldMethod` would need to be updated to support the new number of partial timesteps
3. If using something like a Runge-Kutta scheme, then `EnzoIntegrableUpdate::update_quantities` and `EnzoBfieldMethod::update_all_bfield_components` need to be tweaked since the final fluid properties are based on an average of the properties at the partial timesteps.

I don't it's necessary, but if the ghost zones do need to be updated at partial timesteps that might take some extra work. I'm pretty sure it's possible; there has been talk of using this sort of feature for refreshing particles for stellar feedback and ray-tracing.

I can talk more about this if you want (Philipp also seemed very optimistic that this is possible).